### PR TITLE
Convert fv_dynamics into object

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

To ensure stencils are compiled and storages are allocated only once, we are shifting components from single functions into objects with an initialization and call method. This PR contains our first full chain implementation of this concept from fv_dynamics to xppm/yppm.

## Code changes:

- fv_dynamics, fvtp2d, xppm, yppm, and tracer_2d_1l are converted into objects which do not compile stencils or allocate storages when called.

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] Unit tests are added or updated for non-stencil code changes

Additionally, if this PR contains code authored by new contributors:

- [ ] The names of all the new contributors have been added to CONTRIBUTORS.md

### CODE REVIEW

**Comment by rheacangeo** [comment]
launch jenkins

File: fv3core/stencils/fv_dynamics.py [hunk 1]
```diff
@@ -253,65 +182,254 @@ def wrapup(state, comm: CubedSphereCommunicator):
...
+
+        Args:
+            state: model prognostic state and inputs
+            consv_te: ???
+            do_adiabatic_init: if True, do adiabatic dynamics. Used
+                for model initialization.
+            timestep: time to progress forward in seconds
+            ptop: pressure at top of atmosphere
+            n_split: number of acoustic timesteps per remapping timestep
+            ks: ???
```
Comment (rheacangeo) [inline fv3core/stencils/fv_dynamics.py:317]: this is to set the bottom layer for which rayleigh friction and other rayliegh computations are done

**Review by rheacangeo — COMMENTED** [review]
Exciting!

**Comment by rheacangeo** [comment]
Would you mind also updating examples/standalone/runfile/dynamics.py so that when the performance tests are run they can use this?

**Comment by rheacangeo** [comment]
This is failing because the call to the tracer_2d_1l object no longer takes the comm argument, otherwise, very close!

File: fv3core/decorators.py [hunk 2]
```diff
@@ -67,30 +67,35 @@ def state_inputs(*arg_specs):
...
-            func(types.SimpleNamespace(**namespace_kwargs), *args, **kwargs)
+            namespace = get_namespace(arg_specs, state)
+            func(namespace, *args, **kwargs)
 
         return wrapped
 
     return decorator
 
 
+def get_namespace(arg_specs, state):
```
[the remark this answers is not in the document]
Comment (mcgibbon) [reply fv3core/decorators.py:78]: Yes, I need to use it directly as a function in the new fv_dynamics object.

File: fv3core/stencils/dyn_core.py [hunk 3]
```diff
@@ -190,14 +190,14 @@ def compute(state, comm):
 
     init_step = state.n_map == 1
     end_step = state.n_map == spec.namelist.k_split
-    akap = state.akap
+    akap = constants.KAPPA
```
[the remark this answers is not in the document]
Comment (mcgibbon) [reply fv3core/stencils/dyn_core.py:193]: Might be git silliness.

**Review by mcgibbon — COMMENTED** [review]
Ready for re-review

**Pull request merged**